src/WebJobs.Extensions.OpenAI.CosmosDBNoSqlSearch/CosmosDBNoSqlSearchProvider.cs

```diff
@@ -0,0 +1,348 @@
+        SearchableDocument document,
+        CancellationToken cancellationToken
+    )
+    {
+        // Create cosmos client if not exists in the cache.
+        CosmosClient cosmosClient = this.GetCosmosClient(document.ConnectionInfo!.ConnectionName);
+
+        DatabaseResponse databaseResponse = await cosmosClient
+            .CreateDatabaseIfNotExistsAsync(
+                this.cosmosDBNoSqlSearchConfigOptions.Value.DatabaseName,
+                cancellationToken: cancellationToken
+            )
+            .ConfigureAwait(false);
+
+        VectorEmbeddingPolicy vectorEmbeddingPolicy = new VectorEmbeddingPolicy(
+            [
+                new Embedding
+                {
+                    DataType = this.cosmosDBNoSqlSearchConfigOptions.Value.VectorDataType,
+                    Dimensions = this.cosmosDBNoSqlSearchConfigOptions.Value.VectorDimensions,
+                    DistanceFunction = this.cosmosDBNoSqlSearchConfigOptions
+                        .Value
+                        .VectorDistanceFunction,
+                    Path = this.cosmosDBNoSqlSearchConfigOptions.Value.EmbeddingKey,
+                }
+            ]
+        );
+
+        IndexingPolicy indexingPolicy = new IndexingPolicy
+        {
+            VectorIndexes = new Collection<VectorIndexPath>
+            {
+                new()
+                {
+                    Path = this.cosmosDBNoSqlSearchConfigOptions.Value.EmbeddingKey,
+                    Type = this.cosmosDBNoSqlSearchConfigOptions.Value.VectorIndexType,
+                },
+            },
+        };
+        // Create a container if not exists.
+        ContainerProperties containerProperties = new ContainerProperties(
+            document.ConnectionInfo!.CollectionName,
+            "/id"
+        )
+        {
+            VectorEmbeddingPolicy = vectorEmbeddingPolicy,
+            IndexingPolicy = indexingPolicy,
+        };
+
+        ContainerResponse containerResponse = await databaseResponse
+            .Database.CreateContainerIfNotExistsAsync(
+                containerProperties,
+                cancellationToken: cancellationToken
+            )
+            .ConfigureAwait(false);
+
+        await this.UpsertVectorAsync(cosmosClient, document, cancellationToken);
+    }
+
+    CosmosClient GetCosmosClient(String connectionStringName)
+    {
+        CosmosClient cosmosClient;
+        if (
+            !string.IsNullOrEmpty(connectionStringName)
+            && this.cosmosDBClients.TryGetValue(connectionStringName, out cosmosClient)
+        )
+        {
+            cosmosClient = this.cosmosDBClients[connectionStringName];
+        }
+        else
+        {
+            if (!string.IsNullOrEmpty(connectionStringName))
+            {
+                cosmosClient = this.CreateCosmosClient(connectionStringName, false);
+            }
+            else
+            {
+                cosmosClient = this.CreateCosmosClient(connectionStringName, true);

```

Comment:
this code can only be reached if connectionStringName is null or empty but it is necessary to get the cosmos account URI for managed identity. How does it work for managed identity then?

Reply:
Fixed